app/controllers/shelters_controller.rb

```diff
@@ -78,11 +78,24 @@ def drafts
   # This is the definition of a beautiful hack. 1 part gross, 2 parts simplicity. Does something neat not clever.
   def set_headers
     if(admin?)
-      @columns = Shelter::ColumnNames + Shelter::PrivateFields
-      @headers = Shelter::HeaderNames + Shelter::PrivateFields.map(&:titleize)
+      columns = Shelter::ColumnNames + Shelter::PrivateFields
+      @columns = columns
+      @headers = columns.map(&:titleize)
     else
       @columns = Shelter::ColumnNames
-      @headers = Shelter::HeaderNames
+      @headers = Shelter::ColumnNames.map(&:titleize)
+    end
```

Comment:
This could totally be done separately, but here's a thought:

```ruby
def set_headers
  @columns = Shelter::ColumnNames
  @headers = Shelter::HeaderNames

  if admin?
    # add admin columns
  end

  if params[:action] == 'index'
    # remove index columns
  end

  @headers.map!(&:titleize)
end
```